src/subs/WCSFitRoutine.cpp

```
@@ -0,0 +1,672 @@
+    expo->field = exposures_.fieldNumbers[i];
+    expo->instrument = exposures_.instrumentNumbers[i];
+    expo->airmass = exposures_.airmasses[i];
+    expo->exptime = exposures_.exposureTimes[i];
+    expo->mjd = exposures_.mjds[i];
+    tmpExposures.emplace_back(std::move(expo));
+  }
+  setExposures(std::move(tmpExposures), sysErr, refSysErr);
+
+  // Set Extensions:
+  extensions.reserve(extensionExposureNumbers.size());
+  for (int i = 0; i < extensionExposureNumbers.size(); i++)
+  {
+    unique_ptr<Extension> extn(new Extension());
+    int iExposure = extensionExposureNumbers[i];
+    extn->exposure = iExposure;
+    const Exposure &expo = *exposures[iExposure];
+    extn->device = extensionDevices[i];
+    shared_ptr<astrometry::Wcs> tmpWcs = wcss[i];
+    extn->startWcs = unique_ptr<astrometry::Wcs>(tmpWcs.get());
+    extn->startWcs->reprojectTo(*expo.projection);
+    if (expo.instrument < 0)
+    {
+      // This is the reference catalog:
+      extn->mapName = astrometry::IdentityMap().getName();
+    }
+    else
+    {
+      astrometry::YAMLCollector::Dictionary d;
+      d["INSTRUMENT"] = instruments[expo.instrument]->name;
+      d["DEVICE"] = instruments[expo.instrument]->deviceNames.nameOf(extn->device);
+      d["EXPOSURE"] = expo.name;
+      d["BAND"] = instruments[expo.instrument]->band;
+      // Add Epoch to the dictionary if it is not empty
+      if (!expo.epoch.empty())
+        d["EPOCH"] = expo.epoch;
+      // This will be the name of the extension's final WCS and photometry map:
+      extn->wcsName = d["EXPOSURE"] + "/" + d["DEVICE"];
+      // And this is the name of the PixelMap underlying the WCS, or
+      // the name of the photometric map (for which we do not want the "base" part).
+      extn->mapName = extn->wcsName + "/base";
+      if (!inputYAML.addMap(extn->mapName, d))
+      {
+        cerr << "Input YAML files do not have complete information for map "
+             << extn->mapName
+             << endl;
```

This should probably throw an exception.